recipe-client-addon/lib/AboutStudiesProtocol.jsm

```diff
@@ -0,0 +1,105 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+const {utils: Cu, interfaces: Ci, manager: Cm, results: Cr} = Components;
+Cm.QueryInterface(Ci.nsIComponentRegistrar);
+
+Cu.import("resource://gre/modules/Services.jsm");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+this.EXPORTED_SYMBOLS = ["AboutStudiesProtocol"];
+
+/**
+ * Component definition for the about:studies protocol handler.
+ * Registers a component with the browser that establishes an `about:studies`
+ * protocol handler. Navigating to `about:studies` displays the
+ * `AboutStudies.xml` file.
+ */
+function StudiesProtocolHandler() {}
+StudiesProtocolHandler.prototype = {
+  uri: Services.io.newURI("resource://shield-recipe-client/lib/AboutStudies.xml"),
+  classDescription: "about:studies page module",
+  classID: Components.ID("c7c3dd48-c1cf-4bbf-a5df-69eaf6cb27d9"),
+  contractID: "@mozilla.org/network/protocol/about;1?what=studies",
+  QueryInterface: XPCOMUtils.generateQI([Ci.nsIStudiesProtocolHandler]),
+
+  newChannel: (aURI) => {
```

Comment:
The `a` prefix stands for "argument", and is generally going out of style in Firefox code. This is a nit, but can we not use it?